### PR TITLE
Batch account location lookups for sparse ledger

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon #14520. This adds batching for the location lookups in the sparse ledger conversion function.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-nightly-me

**Comment by mrmr1993** [comment]
!ci-build-me

File: src/lib/merkle_mask/masking_merkle_tree.ml [hunk 1]
```diff
@@ -535,16 +535,44 @@ module Make (Inputs : Inputs_intf.S) = struct
...
     let location_of_account_batch t account_ids =
       assert_is_attached t ;
-      let found_locations, leftover_account_ids =
-        List.partition_map account_ids ~f:(fun account_id ->
-            match self_find_location t account_id with
-            | Some location ->
-                Either.first (account_id, Some location)
-            | None ->
-                Either.second account_id )
+      let account_ids_with_locations_rev, leftover_account_ids_rev =
```
Comment (georgeee) [inline src/lib/merkle_mask/masking_merkle_tree.ml:538]: I suggest using the [version](https://github.com/MinaProtocol/mina/commit/8a5022369d5ae850cd882c3c681bd129eb387822)

Comment (nholland94) [reply src/lib/merkle_mask/masking_merkle_tree.ml:538]: I took your version, with some minor modifications (your version was reversing the resulting list, and had a some misnomers about the parent always being a database).

**Comment by nholland94** [comment]
!ci-build-me

**Review by Sventimir — APPROVED** [review]
I'm a little bit concerned about performance of `self_find_or_batch_lookup`. Does the performance gain of batch lookup justify going over the list of identifiers 3 times here? Is preserving the order of identifiers important? If it's not, then this could be done a little faster.

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by georgeee** [comment]
!ci-build-me

**Review by nholland94 — APPROVED** [review]

**Comment by georgeee** [comment]
!ci-build-me

**Comment by nholland94** [comment]
> I'm a little bit concerned about performance of `self_find_or_batch_lookup`. Does the performance gain of batch lookup justify going over the list of identifiers 3 times here? Is preserving the order of identifiers important? If it's not, then this could be done a little faster.

@Sventimir What's your performance concern exactly? The operation is O(n). Are you concerned about garbage collection pressure from the intermediate tuples?

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-nightly-me

**Comment by deepthiskumar** [comment]
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/875#_

**Pull request merged**